### PR TITLE
[CMOB - 2398] Split favourites

### PR DESCRIPTION
**Pull request by DavitTsiskarashvili**
https://spaceneobank.atlassian.net/browse/CMOB-2398

### CODE REVIEW

File: app/src/main/java/com/movies/common/extensions/ViewExtension.kt [hunk 1]
```diff
@@ -50,4 +53,10 @@ fun ImageView.changeDrawable(@DrawableRes drawableRes: Int) {
 
 fun ImageView.changeDrawableColor(@ColorRes colorRes: Int) {
     imageTintList = ContextCompat.getColorStateList(context, colorRes)
+}
+
+fun EditText.hideKeyboard() {
```
Comment (bchmsl) [inline app/src/main/java/com/movies/common/extensions/ViewExtension.kt:58]: Make this Context extension

File: app/src/main/java/com/movies/presentation/favourite/ui/FavouriteFragment.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    override val layout = R.layout.fragment_favourites
+
+    private val favouriteMovieAdapter by lazy { FavouriteMovieAdapter() }
+
+    override fun onRefresh() = viewModel.fetchFavouriteMovies()
+
+    override fun onDataLoaded(data: List<MovieUIModel>) {
+        favouriteMovieAdapter.submitList(data)
+        binding.emptyListImageView.visibleIf(data.isEmpty())
+        binding.emptyListTextView.visibleIf(data.isEmpty())
```
Comment (bchmsl) [inline app/src/main/java/com/movies/presentation/favourite/ui/FavouriteFragment.kt:29]: You can create extension to hide or show multiple views

**Review by bchmsl — APPROVED** [review]

File: app/src/main/java/com/movies/common/extensions/ViewExtension.kt (hunk 1 above)
Comment (DavitTsiskarashvili) [reply app/src/main/java/com/movies/common/extensions/ViewExtension.kt:58]: done in base gradle PR

File: app/src/main/java/com/movies/presentation/favourite/ui/FavouriteFragment.kt (hunk 2 above)
Comment (DavitTsiskarashvili) [reply app/src/main/java/com/movies/presentation/favourite/ui/FavouriteFragment.kt:29]: done in base gradle PR

**Review by AleksandreGalogreBaramidze — APPROVED** [review]

File: app/src/main/java/com/movies/presentation/favourite/ui/FavouriteFragment.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        }
+        // Navigate to Details Screen
+        favouriteMovieAdapter.onItemClickListener { film ->
+            changeScreen(DetailsFragment(), film.id)
+        }
+        // Navigate to Home Screen
+        binding.navigationButton.leftButtonListener {
+            changeScreen(HomeFragment(), null)
+        }
+    }
```
Comment (etozheluka) [inline app/src/main/java/com/movies/presentation/favourite/ui/FavouriteFragment.kt:52]: Why are you handling navigation in a fragment? Additionally, you have 'null' as the default value in your function. Is it still necessary to pass 'null'?

File: app/src/main/java/com/movies/presentation/home/view_model/HomeViewModel.kt [hunk 4]
```diff
@@ -124,6 +98,7 @@ class HomeViewModel(
         viewModelScope {
             flow.cachedIn(viewModelScope).collect { pagingData ->
                 val mappedData = pagingData.map {
+                    Log.d("giorgi", pagingData.toString())
```
Comment (etozheluka) [inline app/src/main/java/com/movies/presentation/home/view_model/HomeViewModel.kt:101]: giorgi

Comment (DavitTsiskarashvili) [reply app/src/main/java/com/movies/presentation/home/view_model/HomeViewModel.kt:101]: already have deleted in the next PR

File: app/src/main/java/com/movies/presentation/favourite/ui/FavouriteFragment.kt (hunk 3 above)
Comment (DavitTsiskarashvili) [reply app/src/main/java/com/movies/presentation/favourite/ui/FavouriteFragment.kt:52]: already refactored whole navigation logic in the next PR

**Review by etozheluka — APPROVED** [review]

**Review by AnaRusadze — APPROVED** [review]

**Pull request merged**